quantum/process_keycode/process_auto_shift.c

```diff
@@ -225,6 +188,12 @@ void set_autoshift_timeout(uint16_t timeout) { autoshift_timeout = timeout; }
 
 bool process_auto_shift(uint16_t keycode, keyrecord_t *record) {
     if (record->event.pressed) {
+        if (autoshift_flags.in_progress) {
+            autoshift_end(KC_NO, record->event.time, false);
```

Comment:
Excuse my ignorance, but can you explain under what circumstances KC_NO occurs here?

Reply:
That line is to evaluate the current AutoShift key on another down. I used to have a `if (keycode == autoshift_lastkey)` in the `in_progress` section of `autoshift_end` (as testing if it was an AS key would still be true if it was one cancelling another), but as that's gone the `KC_NO` isn't necessary. It does make it clear the line isn't to test the current key, though - I'll add a comment too.

Reply:
Still not required here but the fix for [this](https://github.com/p00ya/qmk_firmware/pull/1#issuecomment-666553392) in custom shift pull means everything in that slot is necessary.